### PR TITLE
fix(#753): API documentation for `unittest.mock`

### LINKED ISSUE #753: TypeError: argument of type '_SentinelObject' is not iterable (`unittest.mock`)
#### Problem Description

`pdoc` fails to generate API documentation for the `unittest.mock` module.

#### Steps to reproduce the behavior:

```sh
pdoc --output-directory /tmp/docs unittest.mock
```
```py
Traceback (most recent call last):
  File "/private/tmp/pdoc/.venv/bin/pdoc", line 8, in <module>
    ...
  File "/Users/bf/.pyenv/versions/3.13.0/lib/python3.13/functools.py", line 1037, in __get__
    val = self.func(instance)
  File "/private/tmp/pdoc/.venv/lib/python3.13/site-packages/pdoc/doc.py", line 259, in members
    taken_from = self._taken_from(name, obj)
  File "/private/tmp/pdoc/.venv/lib/python3.13/site-packages/pdoc/doc.py", line 448, in _taken_from
    if mod and qual and "<locals>" not in qual:
                        ^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type '_SentinelObject' is not iterable
```

#### System Information

```sh
❯ pdoc --version
pdoc: 15.0.0
Python: 3.13.0
Platform: macOS-14.6.1-arm64-arm-64bit-Mach-O
```

### PR DESCRIPTION
**Pull request by bryant-finney**
# Closes #753 

Resolve the `TypeError` by testing for the `__contains__` attribute

### CODE REVIEW

File: pdoc/doc.py [hunk 1]
```diff
@@ -445,7 +445,7 @@ def _taken_from(self, member_name: str, obj: Any) -> tuple[str, str]:
 
         mod = _safe_getattr(obj, "__module__", None)
         qual = _safe_getattr(obj, "__qualname__", None)
-        if mod and qual and "<locals>" not in qual:
+        if mod and hasattr(qual, "__contains__") and "<locals>" not in qual:
```
Comment (mhils) [inline pdoc/doc.py:448]: Do you think
```suggestion
        if mod and isinstance(qual, str) and "<locals>" not in qual:
```
would maybe be more appropriate here?

**Review by mhils — COMMENTED** [review]
Thanks! 🍰

File: pdoc/doc.py (hunk 1 above)
Comment (bryant-finney) [reply pdoc/doc.py:448]: Ah yes, nice catch 👍 (should have checked the signature)

**Review by mhils — APPROVED** [review]
Thanks!

**Pull request merged**